scipy/special/specfun.c

```diff
@@ -0,0 +1,8417 @@
+        b1 = 1.5 - (el + log(x / 2.0));
+        rs = 1.0;
+        r = 1.0;
+        for (k = 2; k <= 50; k++) {
+            r = 0.25*r*(k - 1.0) / (k*k*k)*x*x;
+            rs += 1.0 / k;
+            r2 = r * (rs + 1.0 / (2.0*k) - (el + log(x / 2.0)));
+            b1 += r2;
+            if (fabs(r2 / b1) < 1.0e-12) { break; }
+        }
+        *ttk = e0 - 0.125*x*x*b1;
+    } else {
+        *ttk = 1.0;
+        r = 1.0;
+        for (k = 1; k <= 8; k++) {
+            r = -r / x;
+            *ttk += c[k-1] * r;
+        }
+        rc = x*sqrt(2.0 / (pi * x));
+        *ttk = (*ttk) * exp(-x) / rc;
+    }
+    return;
+}
+
+
+void specfun_ittjya(double x, double *ttj, double *tty) {
+
+    // =========================================================
+    // Purpose: Integrate [1-J0(t)]/t with respect to t from 0
+    //          to x, and Y0(t)/t with respect to t from x to ∞
+    // Input :  x   --- Variable in the limits  ( x ≥ 0 )
+    // Output:  TTJ --- Integration of [1-J0(t)]/t from 0 to x
+    //          TTY --- Integration of Y0(t)/t from x to ∞
+    // =========================================================
+
+    int k, l;
+    double a0, bj0, bj1, by0, by1, e0, b1, g0, g1, px, qx, r, r0, r1, r2, rs, t, vt, xk;
+    const double pi = 3.141592653589793;
+    const double el = 0.5772156649015329;
+
+    if (x == 0.0) {
+        *ttj = 0.0;
+        *tty = -1.0e+300;
+    } else if (x <= 20.0) {
+        *ttj = 1.0;
+        r = 1.0;
+        for (k = 2; k <= 100; k++) {
+            r = -0.25 * r * (k - 1.0) / (k*k*k)*x*x;
+            *ttj += r;
+            if (fabs(r) < fabs(*ttj) * 1.0e-12) { break; }
+        }
+        *ttj *= 0.125*x*x;
+        e0 = 0.5*(pi*pi/6.0 - el*el) - (0.5*log(x / 2.0) + el)*log(x/2.0);
+        b1 = el + log(x / 2.0) - 1.5;
+        rs = 1.0;
+        r = -1.0;
+        for (k = 2; k <= 100; k++) {
+            r = -0.25 * r * (k - 1.0) / (k*k*k)*x*x;
+            rs = rs + 1.0 / k;
+            r2 = r * (rs + 1.0/(2.0 * k) - (el + log(x/2.0)));
+            b1 = b1 + r2;
+            if (fabs(r2) < fabs(b1) * 1.0e-12) { break; }
+        }
+        *tty = 2.0/pi*(e0 + 0.125 * x * x * b1);
+    } else {
+        a0 = sqrt(2.0 / (pi * x));
+        bj0 = 0.0;
+        by0 = 0.0;
+        bj1 = 0.0;
+        for (l = 0; l <= 1; l++) {
+            vt = 4.0 * l * l;
+            px = 1.0;
+            r = 1.0;
+            for (k = 1; k <= 14; k++) {
+                r = -0.0078125*r*(vt-pow((4.0*k-3.0), 2))/(x*k)*(vt-pow((4.0*k-1.0), 2))/((2.0*k-1.0)*x);
+                px += r;
+                if (fabs(r) < fabs(px) * 1.0e-12) { break; }
+            }
+            qx = 1.0;
+            r = 1.0;
+            for (k = 1; k <= 14; k++) {
+                r = -0.0078125*r*(vt-pow((4.0*k-1.0), 2))/(x*k)*(vt-pow((4.0*k+1.0), 2))/((2.0*k+1.0)*x);
+                qx += r;
+                if (fabs(r) < fabs(qx) * 1.0e-12) { break; }
+            }
+            qx = 0.125 * qx * (vt - 1.0) / x;
```

Comment:
```suggestion
            qx = 0.125 *  (vt - 1.0) / x *qx;
```

this one follows a different pattern in the original

https://github.com/scipy/scipy/blob/829d0dd950d63caece04526b3a617bbe2bc24023/scipy/special/specfun/specfun.f#L1201